src/vsr/replica.zig

```diff
@@ -4883,21 +4883,6 @@ pub fn ReplicaType(
             const op_min = if (range) |r| r.op_max + 1 else range_min;
             assert(op_min <= op);
 
-            if (self.op_checkpoint() == 0 and range != null) {
-                // We get here only if we are a backup with a missing/corrupt root op, advancing our
-                // checkpoint mid-repair. Primaries can never have a corrupt root op as repair
-                // ensures a primary's journal is clean before it transitions to .normal status.
-                assert(self.status == .normal);
-                assert(self.backup());
-                assert(self.commit_min == self.op_checkpoint_next_trigger());
-                assert(op_min == 1);
-                assert(range.?.op_max == 0);
-                assert(range.?.op_min == 0);
-                assert(self.journal.faulty.bit(.{ .index = 0 }));
-            } else {
-                assert(op_min <= self.op_repair_min());
```

Comment:
Earlier, we guaranteed a clean, non-faulty log from op_repair_min → self.op, while now we only guarantee clean log from op_checkpoint → self.op. Therefore, we can't assert `assert(op_min <= self.op_repair_min());`.
https://github.com/tigerbeetle/tigerbeetle/blob/ed0a1c73a24b2caa9fbcecf54b52dfbb27ded83d/src/vsr/replica.zig#L8357